tests/test_maintenance.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import io
+import fnmatch
+import unittest
+import unittest.mock
+import subprocess
+from tools.maintenance import maintenance
+
+DESCRIPTION = """
+This is a unit test script for the tools/maintenance/maintenance.py.
+
+Usage (run from top level directory of hpc toolkit):
+python -m unittest 
+"""
+
+# hpc-toolkit-test0 will have nodes that have maintenance
+# hpc-toolkit-test1 will not have nodes that have maintenance
+valid_prj_list = ["hpc-toolkit-test0", "hpc-toolkit-test1"]
+sched_maint_vms = [f"vm_test{x}" for x in range(10)]
+T1 = "2024-10-26T13:25:51-0700"
+T2 = "2024-10-26T13:26:51-0700"
+up_maint_vms = [[sched_maint_vms[0], T1, T2, "TRUE", "SCHEDULED"],
```

Comment:
Add a comment defining the columns you use here:
`# vm_name, start_time, end_time, ...`